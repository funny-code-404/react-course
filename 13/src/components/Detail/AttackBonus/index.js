import PropTypes from "prop-types";

const AttackBonus = ({ data }) => (
	data && 
	<div>
		<p>Attack bonus:</p>
		<ul>
			{data.map((x) => <li key={x}>{x}</li>)}
		</ul>
	</div>
)

AttackBonus.propTypes = {
	data: PropTypes.arrayOf(PropTypes.string),
}

export default AttackBonus;