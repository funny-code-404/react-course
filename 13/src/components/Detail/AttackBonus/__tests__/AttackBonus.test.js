import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import AttackBonus from "..";

configure({ adapter: new Adapter() });

describe('DETAIL ATTACK BONUS', () => {
    const props = {
        data: [
             'Some text...',
        ]
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<AttackBonus {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Component is return 1 content', () => {
        expect(wrapper.find('li')).toHaveLength(1);
    });
    it('Component should render correct data content', () => {
        expect(wrapper.find('li').props().children).toEqual(props.data[0]);
    });
    it('Component should render correct after props changing', () => {
        const newdata = 'NEW';
        wrapper.setProps({ data: [ newdata, ]})
        expect(wrapper.find('li').props().children).toEqual(newdata);
    });
});

// npm run test src/components/Detail/AttackBonus