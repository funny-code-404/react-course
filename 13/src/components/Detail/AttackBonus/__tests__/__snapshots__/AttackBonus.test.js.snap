// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttackBonus to Match Snapshot 1`] = `
<div>
  <p>
    Attack bonus:
  </p>
  <ul>
    <li
      key="Some text..."
    >
      Some text...
    </li>
  </ul>
</div>
`;

exports[`DETAIL ATTACK BONUS to Match Snapshot 1`] = `
<div>
  <p>
    Attack bonus:
  </p>
  <ul>
    <li
      key="Some text..."
    >
      Some text...
    </li>
  </ul>
</div>
`;
